### ISSUE TITLE
fix(tests): test_batchnorm.mojo fails with numerical precision errors

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#2563]
## Problem

`tests/shared/core/layers/test_batchnorm.mojo` fails with numerical precision error:
```
Unhandled exception caught during execution: 0.0 !≈ 1.0 (diff: 1.0)
```

## Context

Discovered when PR #2554 added `-Werror` flag and PR #2555 added test discovery, exposing pre-existing test bugs.

## Root Cause

The test has incorrect numerical expectations for batch normalization operations.

## Action Required

1. Debug the test to identify which assertion is failing
2. Verify the expected values against BatchNorm2d mathematical definition
3. Fix the test with correct expected values or appropriate tolerances

## Related

- Part of test failures blocking PR #2554 (zero-warnings policy)
- Part of test failures blocking PR #2555 (test discovery)
